enclave/start/sgx/start.S

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+//==============================================================================
+//
+// _start():
+//
+//     The default entry point for the enclave image. This entry point jumps
+//     directly to the oe_enter() function in liboecore.
+//
+//==============================================================================
+
+.globl _start
```

Comment:
I am a little lost: why do we have to define `_start` again? The other one is in enclave/sgx/start.S.

Reply:
I was curious too and started looking around.

I believe what has happened is that it's due to a subdirectory inclusion in the CMake files, so it is defined twice if this is an SGX machine, and once if not. The first inclusion also seems to do some registry pushing.

Looking further, however, I believe this has to do with scope limitations in CMake. Looking at sgx/CMakeLists.txt, we can see the subdirectory enclusion of /start if this is an SGX machine.

If we take a look at the library dependencies of start/CMakeLists.txt, we see the library dependency of Start.S. 

Disclaimer: I am not _overly familiar with assembly_, so please take this with a grain of salt. 

But it seems that the `cfi_startproc` needs to be different when being used on an sgx machine, which entails not using the registry pushing when called from the `oestart` library. This is might be the reason why it was chosen to not encapsulate the registry pushing when redefining this.

Which might have resulted in the need to keep `oe_link_enclave` out of the `oe_enter` call, as commented above. If this is true, a slight refactor may be needed but it seems to be the correct way of handling this.